SOURCES/etc/xapi.d/plugins/smartctl.py

```diff
@@ -24,7 +24,9 @@ def get_information(session, args):
     with OperationLocker():
         disks = _list_disks()
         for disk in disks:
-            cmd = run_command(["smartctl", "-j", "-a", disk])
+            cmd = run_command(["smartctl", "-j", "-a", disk], check=False)
+            if cmd['returncode'] != 0 and cmd['returncode'] != 4:
+                raise_plugin_error(cmd['returncode'], "device lacks SMART capability")
```

Comment:
Are you sure any other return code than 0 and 4 means no SMART capability?

In addition to this, a disk without capability shouldn't prevent from getting the data for other disks. Rather than raising, we should probably fill the results for the disk with the contents of stderr.

Reply:
And we should wheck with XO devs what form this could take so that XO can display it.

Reply:
There are many errors and it is a bit field. I'm sure that the one we are seeing in integration with the lack if SMART capability is 4. But as I don't know for others I think it is better to raise them, open a bug and fix. Otherwise we will let some errors passed without knowing that can happens. Currently we are aware of the lack of SMART capability that is error 4 but that's it. Thus as we don't know others errors it is better to treat them as bug IMHO.

Reply:
But the message is wrong. I will fix it.

Reply:
If a user needs to check their disk status and there's one particular disk which causes an unforeseen error, I don't want them to get no information at all. The feature is a troubleshooting one, and we should display whatever information we have for the other disks. Embedding the error inside the output for the disk(s) which cause an unforeseen error is enough to let users contact our support and let us know about the bug.

If we wan't our CI to be more strict, we can always check the contents of the returned data for embedded errors.

Reply:
Ok so I can just remove the check of errors because error is already logged in the output.